src/overlays/actors/ovl_Obj_Jgame_Light/z_obj_jgame_light.c

```diff
@@ -28,29 +39,172 @@ const ActorInit Obj_Jgame_Light_InitVars = {
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_METAL,
+        AT_NONE,
+        AC_ON | AC_HARD | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK2,
+        { 0x00100000, 0x00, 0x00 },
+        { 0xF6CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON | BUMP_HOOKABLE,
+        OCELEM_ON,
+    },
     { 12, 45, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80C15BC0;
-
-extern UNK_TYPE D_060003A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C15474.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C156C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/func_80C15718.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Jgame_Light/ObjJgameLight_Draw.s")
+#include "assets/overlays/ovl_Obj_Jgame_Light/ovl_Obj_Jgame_Light.c"
+
+void ObjJgameLight_Init(Actor* thisx, PlayState* play) {
+    ObjJgameLight* this = THIS;
+    LightInfo* lights = &this->lightInfo;
+
+    Actor_SetScale(&this->actor, 1.0f);
+    func_800B4AEC(play, &this->actor, 50.0f);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B4B50, 1.0f);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Lights_PointGlowSetInfo(lights, this->actor.world.pos.x, (this->actor.world.pos.y + 70.0f), this->actor.world.pos.z,
+                            255, 255, 180, -1);
+    this->lightNode = LightContext_InsertLight(play, &play->lightCtx, &this->lightInfo);
+    Actor_SetFocus(&this->actor, 60.0f);
+    this->actor.colChkInfo.health = 0;
+    this->unk_1B8 = 0;
+    this->unk_1B6 = 0;
+    this->unk_1AC = 0;
+    this->alpha = 0;
+    this->signal = NONE;
+    this->unk_1A8 = 0.0f;
+}
+
+void ObjJgameLight_Destroy(Actor* thisx, PlayState* play) {
+    ObjJgameLight* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+    LightContext_RemoveLight(play, &play->lightCtx, this->lightNode);
+}
+
+void func_80C15474(ObjJgameLight* this, PlayState* play) {
+    u8 temp_a1;
+
+    if (((this->actor.colChkInfo.health & 1) != 0) && (this->unk_1B6 == 0)) {
+        if (this->unk_1AC < 0xA0) {
+            this->unk_1AC += 0x28;
+        } else {
+            this->unk_1AC = 0xC8;
+            this->unk_1B6 = 1;
+        }
+        if (this->unk_1A8 < 0.7f) {
+            this->unk_1A8 += 0.3f;
+        } else {
+            this->unk_1A8 = 1.0f;
+        }
+    } else if ((this->actor.colChkInfo.health & 8) != 0) {
```

Comment:
```suggestion
    } else if (OBJLUPYGAMELIFT_HEALTH_8(&this->actor)) {
```
Would need a new macro